felix/bpf/proxy/topology_test.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2020-2023 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy_test
+
+import (
+	"testing"
+
+	"github.com/projectcalico/calico/felix/bpf/proxy"
+
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func TestShouldAppendTopologyAwareEndpoint(t *testing.T) {
+	testCases := []struct {
+		nodeLabels      map[string]string
```

Comment:
might be handy to add a test description that could be passed into t.Run() so that you would know which test failed if a test fails.

Reply:
Thanks @tomastigera  - added a test description accordingly